src/main/java/uk/gov/hmcts/reform/divorce/orchestration/tasks/SendPetitionerGenericUpdateNotificationEmail.java

```
@@ -0,0 +1,58 @@
+import uk.gov.hmcts.reform.divorce.orchestration.framework.workflow.task.TaskContext;
+import uk.gov.hmcts.reform.divorce.orchestration.framework.workflow.task.TaskException;
+import uk.gov.hmcts.reform.divorce.orchestration.service.EmailService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.D_8_CASE_REFERENCE;
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.D_8_PETITIONER_EMAIL;
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.D_8_PETITIONER_FIRST_NAME;
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.D_8_PETITIONER_LAST_NAME;
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.NOTIFICATION_ADDRESSEE_FIRST_NAME_KEY;
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.NOTIFICATION_ADDRESSEE_LAST_NAME_KEY;
+import static uk.gov.hmcts.reform.divorce.orchestration.domain.model.OrchestrationConstants.NOTIFICATION_CCD_REFERENCE_KEY;
+import static uk.gov.hmcts.reform.divorce.orchestration.tasks.util.TaskUtils.getMandatoryPropertyValueAsString;
+
+@Component
+public class SendPetitionerGenericUpdateNotificationEmail implements Task<Map<String, Object>> {
+
+    private static final String EMAIL_DESC = "Generic Update Notification - Petitioner";
+
+    private final EmailService emailService;
+
+    @Autowired
+    public SendPetitionerGenericUpdateNotificationEmail(EmailService emailService) {
+        this.emailService = emailService;
+    }
+
+    @Override
+    public Map<String, Object> execute(TaskContext context, Map<String, Object> caseData) throws TaskException {
+
+        String petitionerEmail = (String) caseData.get(D_8_PETITIONER_EMAIL);
+
+        if (StringUtils.isNotBlank(petitionerEmail)) {
```

Should never be blank in theory?

---

It shouldn't be but if it is will throw an exception - we could have a log around this if we avoid the exception.

---

@hosainnet  has a good point here, can this be null, if no we should raise an exception here

---

Remember we have the "getMandatoryField" method, which throws the exception for you.